### PR TITLE
fix: attitude callback trigger

### PR DESCRIPTION
**Pull request by andresR8**
After the correct subscription of the topics (#793). When another node publish to the topics (target_attitude & thrust)  the corresponding callback function  `&SetpointAttitudePlugin::attitude_pose_cb` is not triggered.

Apparently the sync objects should be defined as a private property of the plugin and not only locally in the init method.

### CODE REVIEW

File: mavros/src/plugins/setpoint_attitude.cpp [hunk 1]
```diff
@@ -107,6 +107,14 @@ class SetpointAttitudePlugin : public plugin::PluginBase,
 	message_filters::Subscriber<geometry_msgs::PoseStamped> pose_sub;
 	message_filters::Subscriber<geometry_msgs::TwistStamped> twist_sub;
 
+	typedef  message_filters::sync_policies::ApproximateTime<geometry_msgs::PoseStamped, mavros_msgs::Thrust> SyncPoseThrustPolicy;
```
Comment (TSC21) [inline mavros/src/plugins/setpoint_attitude.cpp]: Why not `using` instead of `typedef`?

File: mavros/src/plugins/setpoint_attitude.cpp [hunk 2]
```diff
@@ -107,6 +107,14 @@ class SetpointAttitudePlugin : public plugin::PluginBase,
...
+	typedef  message_filters::Synchronizer<SyncPoseThrustPolicy> SyncPoseThrust;
+	boost::shared_ptr<SyncPoseThrust> sync_thrust;
+
+	typedef  message_filters::sync_policies::ApproximateTime<geometry_msgs::TwistStamped, mavros_msgs::Thrust> SyncTwistThrustPolicy;
+	typedef  message_filters::Synchronizer<SyncTwistThrustPolicy> SyncTwistThrust;
+	boost::shared_ptr<SyncTwistThrust> sync_twist;
```
Comment (TSC21) [inline mavros/src/plugins/setpoint_attitude.cpp]: Organize by types first and then by subject.

File: mavros/src/plugins/setpoint_attitude.cpp [hunk 3]
```diff
@@ -27,11 +27,11 @@
 #include <geometry_msgs/TwistStamped.h>
 #include <mavros_msgs/Thrust.h>
 
+#include <boost/thread.hpp>
+
 namespace mavros {
 namespace std_plugins {
 
-using SyncPoseThrust = message_filters::sync_policies::ApproximateTime<geometry_msgs::PoseStamped, mavros_msgs::Thrust>;
-using SyncTwistThrust = message_filters::sync_policies::ApproximateTime<geometry_msgs::TwistStamped, mavros_msgs::Thrust>;
```
Comment (vooon) [inline mavros/src/plugins/setpoint_attitude.cpp]: Useless change. `using` is fine.

File: mavros/src/plugins/setpoint_attitude.cpp [hunk 4]
```diff
@@ -27,11 +27,11 @@
 #include <geometry_msgs/TwistStamped.h>
 #include <mavros_msgs/Thrust.h>
 
+#include <boost/thread.hpp>
+
```
Comment (vooon) [inline mavros/src/plugins/setpoint_attitude.cpp]: Not needed change.

File: mavros/src/plugins/setpoint_attitude.cpp (hunk 3 above)
Comment (vooon) [reply mavros/src/plugins/setpoint_attitude.cpp]: Just add `Policy` suffix, and then add
```cpp
using SyncPoseThrust = message_filters::Synchronizer<SyncPoseThrustPplicy>;
using SyncTwistThrust = ...;
```

File: mavros/src/plugins/setpoint_attitude.cpp [hunk 5]
```diff
@@ -107,6 +107,14 @@ class SetpointAttitudePlugin : public plugin::PluginBase,
...
 	message_filters::Subscriber<geometry_msgs::TwistStamped> twist_sub;
 
+	typedef  message_filters::sync_policies::ApproximateTime<geometry_msgs::PoseStamped, mavros_msgs::Thrust> SyncPoseThrustPolicy;
+	typedef  message_filters::Synchronizer<SyncPoseThrustPolicy> SyncPoseThrust;
+	boost::shared_ptr<SyncPoseThrust> sync_thrust;
+
+	typedef  message_filters::sync_policies::ApproximateTime<geometry_msgs::TwistStamped, mavros_msgs::Thrust> SyncTwistThrustPolicy;
+	typedef  message_filters::Synchronizer<SyncTwistThrustPolicy> SyncTwistThrust;
+	boost::shared_ptr<SyncTwistThrust> sync_twist;
+
```
Comment (vooon) [inline mavros/src/plugins/setpoint_attitude.cpp]: Here you need only this two lines:
```cpp
std::unique_ptr<SyncPoseThrust> sync_pose;
std::unique_ptr<SyncTwistThrust> sync_twist;
```

**Comment by vooon** [comment]
@andresR8 thank you. Patch applied manually.

**Comment by andresR8** [comment]
@vooon I've tested your commit (mavros 0.21.2) and I cannot receive the corresponding callback `attitude_pose_cb`

**Pull request closed without merging**